Fix #2379: Make PreImageRevealInterval configurable

Geod24 opened the pull request:
Required a slight refactoring otherwise I would have had to change the interface of `getNextPreImage`, which proved very ugly and inconvenient.

Geod24 committed:
EnrollMan: Pass a `ValidatorConfig` directly to the constructor

While this goes against the SoC previously established (don't import `node` from `consensus`),
it is necessary in order to make `PreImageRevealCycle` configurable without the change leaking
through the interface.
It also greatly simplify the rest of the code.

Geod24 committed:
Fix #2379: Make `PreImageRevealInterval` configurable

It is now a `validator` config option.

Geod24 committed:
Move SCP-related config into its own module

This makes exclusion in dub easier, as previously agora.node.Config
imported SCP, leading to a cascade of dependencies for other configurations.

Geod24 commented:
Added to the example config, originally missed it.

linked0 commented on source/agora/consensus/protocol/Config.d:
```diff
@@ -0,0 +1,100 @@
+
+/*******************************************************************************
+
+    Convert a QuorumConfig to the SCPQorum which the SCP protocol understands
```
```suggestion
    Convert a QuorumConfig to the SCPQuorum which the SCP protocol understands
```

linked0 commented on source/agora/consensus/protocol/Config.d:
```diff
@@ -0,0 +1,100 @@
+
+/*******************************************************************************
+
+    Convert an SCPQorum to a QuorumConfig
```
```suggestion
    Convert an SCPQuorum to a QuorumConfig
```

linked0 commented on source/agora/consensus/protocol/Config.d:
```diff
@@ -0,0 +1,100 @@
+    Convert an SCPQorum to a QuorumConfig
+
+    Params:
+        scp_quorum = the quorum config
```
```suggestion
        scp_quorum = `SCPQuorumSet` instance
```

linked0 commented on source/agora/consensus/protocol/Config.d:
```diff
@@ -0,0 +1,100 @@
+        scp_quorum = the quorum config
+
+    Returns:
+        `SCPQuorumSet` instance
```
```suggestion
        a quorum config
```

linked0 reviewed (commented):
Two typos and there were incorrect comments originally. It should be corrected in this PR.

Geod24 committed:
SCP quorum config: Fix documentation

This was spotted in the previous commit, which moved the code and documentation.
However, in order to avoid confusing 'git diff' heuristic for moved code,
the fixes are put in a separate commit.

Geod24 commented:
Thanks @linked0 , I went ahead and applied your suggestions and more.

linked0 reviewed (approved):
LGTM.
This makes testing code simpler.
